### PR TITLE
feat: added forgot functionality

### PR DESCRIPTION
**Pull request by Girishbari**
## Related Issue

Added : #106 
## Description

Added forgot functionality using 2 method API's createRecovery and updatePassword of appwrite and also introduced two new component called as forgot and updatepassword in project
## Screenshots

image

### CODE REVIEW

**Review by Aditi-Dahiya — APPROVED** [review]
Good work done @Girishbari,
Congratulations! 🎉🌟 Sit and relax, you've made your contribution to the Palettegram project. 
I have reviewed the PR and it looks good to me @Sanchitbajaj02 for merging without any conflict. 
Keep contributing🚀
All the best 👍🏻

**Comment by Sanchitbajaj02** [comment]
Hi @Girishbari how much .yarn you've pushed in this branch? And how is it even possible 😅. Can you please delete that.

File: src/app/(auth)/forgot/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  keywords:
+    "Palettegram, Appwrite, Color Designer, Palettegram social media, behance, dribbble, made for designers",
+
+  alternates: {
+    canonical: "https://palettegram.vercel.app/login",
+  },
+};
+
+export default function LoginLayout({ children }: { children: React.ReactNode }) {
```
Comment (Sanchitbajaj02) [inline src/app/(auth)/forgot/layout.tsx:16]: Please change the `LoginLayout` to `ForgotLayout`

File: src/app/(auth)/forgot/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Metadata } from "next";
+
+export const metadata: Metadata = {
+  title: "Login | Palettegram - from professionals by professionals",
```
Comment (Sanchitbajaj02) [inline src/app/(auth)/forgot/layout.tsx:4]: The route will be Forgot Password |... instead of Login |

File: src/app/(auth)/verify/layout.tsx [hunk 3]
```diff
@@ -1,15 +1,15 @@
 import { Metadata } from "next";
 
 export const metadata: Metadata = {
-  title: "Verify | Palettegram - from professionals by professionals",
+  title: "ForgotPassword | Palettegram - from professionals by professionals",
```
Comment (Sanchitbajaj02) [inline src/app/(auth)/verify/layout.tsx:4]: This verify route is for verify users, please revert it back

File: src/backend/appwrite.config.ts [hunk 4]
```diff
@@ -17,7 +17,8 @@ const usersCollection = String(process.env.NEXT_PUBLIC_USER_COLLECTION);
 const postsCollection = String(process.env.NEXT_PUBLIC_POSTS_COLLECTION);
 const bookmarksCollection = String(process.env.NEXT_PUBLIC_BOOKMARKS_COLLECTION);
 const bucketStorage = String(process.env.NEXT_PUBLIC_BUCKET_ID);
+const baseURL = String(process.env.NEXT_PUBLIC_BASE_URL)
 
-export { appwriteClient, account, db, storage, ID, Query };
+export { appwriteClient, account, db, storage, ID, Query, baseURL };
```
Comment (Sanchitbajaj02) [inline src/backend/appwrite.config.ts:22]: Do we not getting this in string format??

File: src/backend/auth.api.ts [hunk 5]
```diff
@@ -117,6 +117,50 @@ const loginUser = async (userData: any) => {
...
+ * @param {string} userData
+ * @returns {Object} response
+ */
+const forgotpassword = async (email: string) => {
+  try {
+    // console.log("login:", userData?.email, userData?.password);
+    if (!email) {
+      throw new Error("email is needed ");
+    }
+    const response = await account.createRecovery(email, `http://localhost:3000/updatepassword`);
```
Comment (Sanchitbajaj02) [inline src/backend/auth.api.ts:132]: Please don't hard code the URL

File: src/backend/auth.api.ts [hunk 6]
```diff
@@ -117,6 +117,50 @@ const loginUser = async (userData: any) => {
...
+    throw new Error(error.message);
+  }
+};
+
+/**
+ * @description Update user's password
+ * @param {Object} userData
+ * @returns {Object} response
+ */
+const updatepassword = async (userData: any) => {
```
Comment (Sanchitbajaj02) [inline src/backend/auth.api.ts:145]: Please handle the type of userData

**Review by Sanchitbajaj02 — CHANGES_REQUESTED** [review]

File: src/backend/auth.api.ts (hunk 5 above)
Comment (Girishbari) [reply src/backend/auth.api.ts:132]: I too don't want it to however whenever I use baseURL from or proccess.env.NEXT_PUBLIC_BASE_URL, I am getting this error
`Invalid url param: URL host must be one of: localhost, cloud.appwrite.io`

File: src/backend/appwrite.config.ts (hunk 4 above)
Comment (Girishbari) [reply src/backend/appwrite.config.ts:22]: I am getting this is in string format

**Comment by Girishbari** [comment]
Please ignore this request, I have submitted another #137

**Pull request closed without merging**